sw/otbn/code-snippets/rsa.s

```diff
@@ -33,8 +34,23 @@ rsa_encrypt:
 rsa_decrypt:
   jal      x1, modload
   jal      x1, modexp
+  jal      x1, cp_work_buf
   ecall
 
+/**
+ * Copy the contents of work_buf onto inout
+ *
+ * clobbered registers: x3, x4
+ */
+cp_work_buf:
+  la  x3, work_buf
+  la  x4, inout
+  /* The buffers are 512 bytes long, which we can load/store with
+     sixteen 256b words. */
+  loopi 16, 2
```

Comment:
Technically we would habe the number of limbs available and could think of making the loop iterations depend on that. But effect on perf. will be very minor, so probably better to keep it generic.